File: devops-scm-provider/devops-scm-provider-svn/devops-scm-provider-svn-common/src/main/java/com/tencent/devops/scm/provider/svn/common/TmatesoftBasicAuthenticationManager.java
package com.tencent.devops.scm.provider.svn.common;

import org.tmatesoft.svn.core.auth.BasicAuthenticationManager;
import org.tmatesoft.svn.core.auth.SVNAuthentication;
import org.tmatesoft.svn.core.io.SVNRepository;

public class TmatesoftBasicAuthenticationManager extends BasicAuthenticationManager {

    public TmatesoftBasicAuthenticationManager(SVNAuthentication[] authentications) {
        super(authentications);
    }

    @Override
    public int getConnectTimeout(SVNRepository repository) {
        int connectTimeout = super.getConnectTimeout(repository);
        connectTimeout = Math.min(connectTimeout, 30 * 1000);
        if (connectTimeout <= 0) {
            connectTimeout = 30 * 1000;
        }
        return connectTimeout;
    }

    @Override
    public int getReadTimeout(SVNRepository repository) {
        int readTimeout = super.getReadTimeout(repository);
        readTimeout = Math.min(readTimeout, 1800 * 1000);
        if (readTimeout <= 0) {
            readTimeout = 1800 * 1000;
        }
        return readTimeout;
    }
}

File: devops-scm-provider/devops-scm-provider-svn/devops-scm-provider-svn-common/src/main/kotlin/com/tencent/devops/scm/provider/svn/common/TmatesoftBasicAuthenticationManager.kt
package com.tencent.devops.scm.provider.svn.common

import org.tmatesoft.svn.core.auth.BasicAuthenticationManager
import org.tmatesoft.svn.core.auth.SVNAuthentication
import org.tmatesoft.svn.core.io.SVNRepository
import kotlin.math.min

class TmatesoftBasicAuthenticationManager(authentications: Array<SVNAuthentication>) : 
    BasicAuthenticationManager(authentications) {

    override fun getConnectTimeout(repository: SVNRepository?): Int {
        var connectTimeout = super.getConnectTimeout(repository)
        connectTimeout = min(connectTimeout, 30 * 1000)
        if (connectTimeout <= 0) {
            connectTimeout = 30 * 1000
        }
        return connectTimeout
    }

    override fun getReadTimeout(repository: SVNRepository?): Int {
        var readTimeout = super.getReadTimeout(repository)
        readTimeout = min(readTimeout, 1800 * 1000)
        if (readTimeout <= 0) {
            readTimeout = 1800 * 1000
        }
        return readTimeout
    }
}